packages/minimongo/cursor.js

```diff
@@ -379,7 +361,17 @@ export default class Cursor {
 
     // run the observe callbacks resulting from the initial contents
     // before we leave the observe.
-    await this.collection._observeQueue.drain();
+    const isReadyPromise = this.collection._observeQueue.drain();
+
+    const isLocalCollection =
+      Meteor.isClient || this.collection.constructor.name === 'LocalCollection'; // Unamed collections
+
+    if (isLocalCollection) handle.isReady = true;
+    else isReadyPromise.then(() => (handle.isReady = true));
+
+    handle.isReadyPromise = Meteor.isClient
+      ? Promise.resolve()
+      : isReadyPromise;
```

Comment:
The `_observeQueue` is either `Meteor._SynchronousQueue` or `Meteor._AsynchronousQueue`. The latter's `drain` returns `Promise<void>`, but the former's returns `void`. That means, this code could look as follows:

```ts
const drainResult = this.collection._observeQueue.drain();
if (drainResult instanceof Promise) {
  handle.isReadyPromise = drainResult;
  drainResult.then(() => (handle.isReady = true));
} else {
  handle.isReady = true;
  handle.isReadyPromise = Promise.resolve();
}
```